f8a_worker/workers/graph_populator.py

```diff
@@ -0,0 +1,242 @@
+            details = input_json.get('analyses').get('metadata', {}).get('details', [])
+            if details and details[0]:
+                declared_licenses = []
+                if details[0].get('declared_licenses'):
+                    # list of license names
+                    declared_licenses = details[0]['declared_licenses']
+                elif details[0].get('declared_license'):
+                    # string with comma separated license names
+                    declared_licenses = details[0]['declared_license'].split(',')
+
+                prp_version += " ".join(map(lambda x: "ver.property('declared_licenses', '" + cls.str_value_cleaner(x) \
+                                                      + "');", declared_licenses))
+                # Create License Node and edge from EPV
+                for lic in declared_licenses:
+                    prp_version += "lic = g.V().has('lname','" + lic + "').tryNext()." \
+                                   "orElseGet{graph.addVertex('vertex_label', 'License', 'lname', '" + lic + "', " \
+                                   "'last_updated'," + str(time.time()) + ")};" \
+                                   "g.V(ver).out('has_declared_license').has('lname', '" + lic + "').tryNext()." \
+                                   "orElseGet{ver.addEdge('has_declared_license', lic)};"
+
+        str_version = drop_prop + str_version + prp_version if prp_version else ''
+
+        return str_version
+
+    @classmethod
+    def construct_package_query(cls, input_json):
+        pkg_name_tokens = re.split('\W+', input_json.get('package'))
+        prp_package = ""
+        str_package = "pkg = g.V().has('ecosystem', ecosystem).has('name', pkg_name).tryNext()" \
+                      ".orElseGet{graph.addVertex('ecosystem', ecosystem, 'name', pkg_name, " \
+                      "'vertex_label', 'Package')};" \
+                      "pkg.property('last_updated', " + str(time.time()) + ");"
+
+        latest_version = input_json.get('latest_version') or ''
+        if latest_version:
+            prp_package += "pkg.property('latest_version', '" + latest_version + "');"
+
+        # Get Github Details
+        if 'github_details' in input_json.get('analyses', {}):
+            gh_details = input_json.get('analyses').get('github_details').get('details', {})
+            gh_prs_last_year_opened = str(gh_details.get('updated_pull_requests', {}).get('year', {}).get('opened', -1))
+            gh_prs_last_month_opened = str(gh_details.get('updated_pull_requests', {}).get('month', {}).get('opened', -1))
+            gh_prs_last_year_closed = str(gh_details.get('updated_pull_requests', {}).get('year', {}).get('closed', -1))
+            gh_prs_last_month_closed = str(gh_details.get('updated_pull_requests', {}).get('month', {}).get('closed', -1))
+            gh_issues_last_year_opened = str(gh_details.get('updated_issues', {}).get('year', {}).get('opened', -1))
```

Comment:
Why is default value here '-1'? I know that it was here before, but what is a reasoning behind this value.

Reply:
just a way to identify that we do not have corresponding github metrics collected

Reply:
I figured this out, but I am asking what is it -1.

Reply:
All those github metrics are Integer in graph. So `-1` works on behalf of `null` .